### ISSUE TITLE
Rule proposal: Prefer `VoidFunction`

### ISSUE THREAD

**Issue by jakeleventhal** [typescript-eslint/typescript-eslint#11250]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+new+plugin+rule%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
- [x] My proposal is not a "formatting rule"; meaning it does not just enforce how code is formatted (whitespace, brace placement, etc).
- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

Have a `--fix`able rule that prefers the TS `VoidFunction` over `() => void` when defining a type.

### Fail Cases

```TypeScript
type SomeFunc = () => void;
```

### Pass Cases

```TypeScript
type SomeFunc = VoidFunction;
```

**Comment by bradzacher** [comment]
Generally now-a-days we don't look to ship rules for stylistic preferences unless there's a really, really good motivating reason a codebase should standardise or if the community is really rallied behind a stylistic decision.

For niche cases like this where it's a purely stylistic personal preference that is also very niche - we don't look to cover it with our plugin.

Instead you can use general purpose rules to ban the undesired code as per this FAQ: https://typescript-eslint.io/troubleshooting/faqs/general#how-can-i-ban-specific-language-feature
